flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/AggregateUtil.scala

```diff
@@ -494,7 +494,9 @@ object AggregateUtil extends Enumeration {
 
       case DATE => DataTypes.INT
       case TIME_WITHOUT_TIME_ZONE => DataTypes.INT
-      case TIMESTAMP_WITHOUT_TIME_ZONE => DataTypes.BIGINT
+      case TIMESTAMP_WITHOUT_TIME_ZONE =>
+        val dt = argTypes(0).asInstanceOf[TimestampType]
+        DataTypes.TIMESTAMP(dt.getPrecision).bridgedTo(classOf[SqlTimestamp])
```

Comment:
This is a little performance sensitive, because it relates de/serialization.
If the precision is less than 3, we can use BIGINT to have a better performance. 

Could you improve this a bit?  Or create a issue and TODO for it.